### PR TITLE
feat!: Update deprecated endpoints in github/action_variables.go

### LINKED ISSUE #3103: Update deprecated endpoints in github/action_variables.go
```
$ ./script/metadata.sh update-go
could not find operation "GET /repositories/{repository_id}/environments/{environment_name}/variables" in openapi_operations.yaml
could not find operation "GET /repositories/{repository_id}/environments/{environment_name}/variables/{name}" in openapi_operations.yaml
could not find operation "POST /repositories/{repository_id}/environments/{environment_name}/variables" in openapi_operations.yaml
could not find operation "PATCH /repositories/{repository_id}/environments/{environment_name}/variables/{name}" in openapi_operations.yaml
could not find operation "DELETE /repositories/{repository_id}/environments/{environment_name}/variables/{name}" in openapi_operations.yaml
```

The official GitHub v3 API documentation has moved the endpoints above and now needs updating.

Detailed discussion around this can be traced to comments under #3102

### PR DESCRIPTION
**Pull request by HariCharan-001**
Fixes:
#3103 

BREAKING-CHANGE: The following endpoints now take `owner` and `repo` names instead of an integer repo ID:
* ActionsService.ListEnvVariables
* ActionsService.GetEnvVariable
* ActionsService.CreateEnvVariable
* ActionsService.UpdateEnvVariable
* ActionsService.DeleteEnvVariable

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @HariCharan-001 !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by HariCharan-001** [comment]
Hi @gmlewis 
The second **approval** is still pending. Just letting you know, in case.

Can you tag anyone you know ?

**Comment by gmlewis** [comment]
I try to not bother volunteers since this repo has a rather high traffic of PRs (in my opinion).
Feel free to ask any developer you know to provide a code review.

**Comment by gmlewis** [comment]
This is holding up other PRs like #3109.
Merging.

**Pull request merged**
